calico/etcddriver/driver.py

```diff
@@ -512,11 +532,18 @@ def _handle_next_watcher_event(self, resync_in_progress):
         :raises FelixWriteFailed:
         :raises ResyncRequested:
         """
+        # PLW: What happens to the watcher thread if stop() gets called? Or does
+        # it plough on happily until the process goes down?
         if self._watcher_queue is None:
             raise WatcherDied()
         while not self._stop_event.is_set():
             # To make sure we always make progress, only trigger a new resync
             # if we're not in the middle of one.
+            # PLW: This means that when we get a kick to do a resync, we stop
+            # watching for a bit, so could miss an update or two. I think we
+            # should try to close that window, even though it's quite small.
```

Comment:
We start the new watcher from the new snapshot index so you're right, we could drop a few events while we start up the new snapshot (but then we'll pick them up in the snapshot processing).

I'm wondering if the periodic resync gives any value now; reworking this to avoid killing the watcher would be tricky due to corner cases where the new snapshot is loaded from a different etcd node than the watcher.